### ISSUE TITLE
`useLoaderData()` incorrectly hides properties with type `Record<string, any>`

### ISSUE THREAD

**Issue by iamriot-dev** [remix-run/remix#7446]
### What version of Remix are you using?

2.0.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```js
export async function loader() {
  const myObj: Record<string, any> = { foo: "bar" };

  return json({
    myObj,
    baz: "example"
  });
}

export default function Page() {
  const loaderData = useLoaderData<typeof loader>();
  // the type of loaderData is:
  // JsonifyObject<{ baz: string; }>

  loaderData.baz; // no issues
  loaderData.myObj; // Error! the property actually still exists but is missing from type

  return null;
}
```

### Expected Behavior

`loaderData.myObj` should appear in the type info of `loaderData` and reflect it's actual type.

### Actual Behavior

`loaderData.myObj` has been erased from `loaderData`'s type. This same issue occurs with `useActionData()`. I believe it has something to do with a change in the `SerializeFrom` type, as this issue did not occur on version `1.19.x`, only after upgrading to `2.0.0`.

For context, I ran into this issue while dehydrating prefetched data for `trpc-swr`.  The `dehydrate` function returns an object with type `Record<string, any>`.

**Comment by MichaelDeBoey** [comment]
Duplicate of #7246
